pkg/inventory/inventory.go

```diff
@@ -0,0 +1,191 @@
+package inventory
+
+import (
+	"sync"
+
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Inventory keeps track of Build object details, on which it can find objects that match the
+// repository URL and trigger rules.
+type Inventory struct {
```

Comment:
This is not a bad thing to have for an initial proof of concept. I am concerned about scalability if we are keeping data in memory for search/lookup purposes. I think we need a roadmap item for scaling the Inventory.